### PR TITLE
Add FDP changes in xNVMe

### PR DESCRIPTION
**Pull request by ankit-sam**
This series adds FDP support in xNVMe.
The initial 2 commits modify the log pages and set / get feature commands by adding a prep function. This is done to avoid changes to the existing api so that it does not become too bulky.
The next 4 commit introduces the required changes in spec, tools, cli and api.

This series is missing changes to toolbox (test) which I will update once Qemu changes are upstream

### CODE REVIEW

**Comment by karlowich** [comment]
This is looking good.
I have fixed the issue causing the test to fail here: #237

File: include/libxnvme_adm.h [hunk 1]
```diff
@@ -100,6 +114,31 @@ int
...
+xnvme_prep_adm_gfeat(struct xnvme_cmd_ctx *ctx, uint32_t nsid, uint8_t fid, uint8_t sel);
+
+/**
+ * Prepare NVMe Set Feature (sfeat) command
+ *
+ * @param ctx Pointer to ::xnvme_cmd_ctx
+ * @param nsid Namespace identifier
+ * @param fid Feature identifier (see NVMe 1.3; Figure 84)
+ * @param feat Structure defining feature attributes
+ * @param save
```
Comment (karlowich) [inline include/libxnvme_adm.h:136]: I think an explanation could be good here. Maybe something like:
"specify that the controller shall save the attribute"

Comment (ankit-sam) [reply include/libxnvme_adm.h:136]: Sure, will update this

**Review by safl — APPROVED** [review]
Thanks @ankit-sam, this LGTM!

**Pull request merged**